From 43116a83537e3d76328d73800121d81648caffb0 Mon Sep 17 00:00:00 2001
From: Hannes Reinecke <hare@suse.de>
Date: Thu, 28 Jun 2018 12:06:22 +0200
Subject: [PATCH] sbitmap: check for valid bitmap in sbitmap_for_each
References: bsc#1090435
Patch-Mainline: No, SLE12-SP3 specific

When iterating across a sbitmap we should be checking if the
bitmap itself is allocations; otherwise we should just return.

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 include/linux/sbitmap.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/sbitmap.h b/include/linux/sbitmap.h
index 80453d5f0661..c3b191de64b7 100644
--- a/include/linux/sbitmap.h
+++ b/include/linux/sbitmap.h
@@ -209,6 +209,8 @@ static inline void sbitmap_for_each_set(struct sbitmap *sb, sb_for_each_fn fn,
 {
 	unsigned int i;
 
+	if (!sb->map)
+		return;
 	for (i = 0; i < sb->map_nr; i++) {
 		struct sbitmap_word *word = &sb->map[i];
 		unsigned int off, nr;
-- 
2.12.3

